--- rarcrack-0.2/rarcrack.h.orig	2007-10-18 17:18:12.000000000 +0200
+++ rarcrack-0.2/rarcrack.h	2011-01-15 21:15:11.415570956 +0100
@@ -31,10 +31,10 @@
 
 char default_ABC[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
 
-const char CMD_DETECT[] = "file -i -b %s"; //this command return what is the file mime type
+const char CMD_DETECT[] = "file --mime-type -b %s"; //this command return what is the file mime type
 
 const char* TYPE[] = {"rar",		"7z",		"zip", ""}; //the last "" signing this is end of the list
-const char* MIME[] = {"application/x-rar", "application/octet-stream", "application/x-zip", ""};
+const char* MIME[] = {"application/x-rar", "application/x-7z-compressed", "application/zip", ""};
 const char* CMD[] = {"unrar t -y -p%s %s 2>&1", "7z t -y -p%s %s 2>&1", "unzip -P%s -t %s 2>&1", ""};
 
 #define PWD_LEN 100
